tinygrad/shape/view.py

```diff
@@ -15,6 +15,58 @@ def strides_for_shape(shape:Tuple[int, ...]) -> Tuple[int, ...]:
   for d in reversed(shape[1:]): strides.append(d*strides[-1])
   return filter_strides(shape, tuple(reversed(strides)))
 
+@functools.lru_cache(maxsize=None)
+def to_shape_strides(shape:Tuple[int, ...], strides:Tuple[int, ...], mask:Optional[Tuple[Tuple[int, int], ...]] = None) -> Tuple[Tuple[int, int, int], ...]:
+  assert len(shape) == len(strides)
+  state = 1 if mask and strides[0] == 0 and mask[0][1] - mask[0][0] == 1 and shape[0] != 1 else 0
```

Comment:
some comments would be helpful here, what are state 0,1,2?

Reply:
done.